_posts/2019-10-4-manual-testing-how-to-become-a-better-tester-of-your-own-code.markdown

```diff
@@ -0,0 +1,45 @@
+---
+layout: post
+title:  "Manual Testing: How to become a better tester of your own code"
+date: 2019-10-4 11:00:00
+categories: ["software-development", "software-quality"]
+author: abizzinotto
+---
+
+Manual testing is a necessary part of software development and quality assurance. And although it's important to have a dedicated tester in your team, you as a developer can also help speed up the QA, and thus the software development process, by becoming a better manual tester of your own code.
+
+But how to do that? I'll cover 4 simple points that will help you get there!
+
+<!--more-->
+
+Manual testing is a necessary part of software development and quality assurance. It’s true that it is prone to errors, after all, every human activity is, and can also be time-consuming and boring. It’s also true that automated testing reduces testing time a great deal and is more reliable. But automated tests can only go so far. Human intuition and curiosity cannot be replaced by an automated task.
+
+Therefore, it is important to have a software tester in your team who knows not only the application but also the end user well. This person will be the “official” manual tester and will find those edge cases for you. But, as a developer, it is also important for you to stay on top of things and manual test your own code before moving it to Quality Assurance. This way, you can ensure your code exactly what it was intended to do.
+
+Here at OmbuLabs we follow some principles and guidelines to manual test our own code before it goes to QA. In this post, we’ll walk you through these.
+
+##It all starts with the user story
+
+Here at OmbuLabs we work with user stories. And every user story has a definition and a scope. Therefore, we have a bug that needs to be fixed, a feature that needs to be added, a chore that needs to be performed, and other situations specific to the project. What all of these have in common is they present a problem to be solved. As such, there is a specific result or outcome that is expected. 
+
+Now, we have automated tests in place to make sure a feature does what it needs and produces the expected outcome. But there are aspects that need manual testing. How does it look on the screen? What if the user decides to follow an alternate path to get the same result?
+
+When manual testing your own code, always go back to the user story and go through the intended flow. Does everything work as expected? Do you see exactly what you’re supposed to see? Is it easy enough for a non-technical person to follow those steps? Where could errors happen?
```

Comment:
```suggestion
When manually testing your own code, always go back to the user story and go through the intended flow. Does everything work as expected? Do you see exactly what you’re supposed to see? Is it easy enough for a non-technical person to follow those steps? Where could errors happen? What are the edge cases that a non-technical user could try? 
```

Not sure if you are interested in this, but it is so true: https://twitter.com/brenankeller/status/1068615953989087232?s=20

Reply:
Hahaha it's very true! Blog post 2 is about becoming a better manual tester for your team. Maybe I can use this there? I'm not sure how copyright would work in this case though.

Reply:
@ABizzinotto I think you can just embed the tweet in the article which takes care of showing proper credit. 😄

Reply:
@etagwerker awesome! Just did :) I also commited your suggestions to the article. Is there anything else I need to do here?